Core/Net/NetFileCache.cs

```diff
@@ -215,6 +272,88 @@ public string GetCachedZip(Uri url, bool test_data = false)
         }
 
         /// <summary>
+        /// Moves the old cache folder to a new one.
+        /// </summary>
+        /// <param name="newPath">The path to move to.</param>
+        /// <param name="move"></param>
+        public bool MoveDefaultCache(string newPath, bool move = false)
+        {
+            if (string.IsNullOrWhiteSpace(newPath))
+            {
+                return false;
+            }
+
+            // Create the new path if it doesn't exists
+            if (!Directory.Exists(newPath))
+            {
+                Directory.CreateDirectory(newPath);
+            }
+
+            // Check that we have list permission
+            try
+            {
+                Directory.GetFiles(newPath);
+            }
+            catch (UnauthorizedAccessException)
+            {
+                return false;
+            }
+
+            string testFilePath = Path.Combine(newPath, "CKAN_TestFile");
+
+            // Check that we have read/write permission
+            try
+            {
+                FileStream a = File.Create(testFilePath);
+                a.WriteByte(0);
+                a.Close();
+
+                a = File.OpenRead(testFilePath);
+                a.ReadByte();
+                a.Close();
+
+                File.Delete(testFilePath);
+            }
+            catch (UnauthorizedAccessException)
+            {
+                return false;
+            }
+
+            // TODO: Start a new transaction
+
+            // Get a list of all the files in the old cache and move them over
+            var files = Directory.GetFiles(cachePath);
+
+            foreach (string file in files)
+            {
+                string newFilePath = Path.Combine(newPath, Path.GetFileName(file));
```

Comment:
If we're combining multiple old caches, there's a good chance there will be collisions, as eliminating redundant downloads and disk space is one of the reasons this project was requested. We should check whether `newFilePath` already exists and do something intelligent if it does. Maybe print an error, maybe just delete it if we're moving and the file contents are the same.